com.unity.visualeffectgraph/Editor/GraphView/Views/VFXView.cs

```diff
@@ -1449,7 +1449,10 @@ void OnSave()
             foreach (var graph in graphToSave)
             {
                 if (EditorUtility.IsDirty(graph) || UnityEngine.Object.ReferenceEquals(graph, controller.graph))
+                {
+                    graph.UpdateSubAssets();
```

Comment:
The missing UpdateSubAssets reminds me an old debug, see this [conversation](https://unity.slack.com/archives/C47JPNNTZ/p1612430874062800?thread_ts=1612430821.062300&cid=C47JPNNTZ)
At that time, I experimented [a C++ workaround](https://ono.unity3d.com/unity/unity/pull-request/120230/_/graphics/vfx/fix/1307562-unexpected-asset-modification#modulesvfxeditorpublicvisualeffectresourcecpp_n366)
For sure, even this bug : https://fogbugz.unity3d.com/f/cases/1322919/ isn't reproducible anymore, this change will avoid unexpected orphan references saved in VisualEffectAsset.